### ISSUE TITLE
TypeError: __init__() got an unexpected keyword argument 'port'

### ISSUE THREAD

**Issue by patatesboy** [unifi-toolbox/unifi-protect-video-downloader#49]
Hello,

I am using a UDMP, unifi protect version 1.17.3, unifi OS version 1.9.0. 

I was able to connect and download just fine with the `events` and `download` options, but when I use `sync`, I get the following error.

I enter:
`docker run --volume path/to/destination:/downloads unifitoolbox/protect-archiver sync --address 192.168.1.1  --username fakeuser --password fakepassword /downloads 
`

`Traceback (most recent call last):
  File "/usr/local/bin/protect-archiver", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/site-packages/protect_archiver/cli/__init__.py", line 20, in main
    cli.main()
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/protect_archiver/cli/sync.py", line 87, in sync
    client = ProtectClient(
TypeError: __init__() got an unexpected keyword argument 'port'`

So I figured it might be going to the wrong port, but I get the same error even if I attach the `--port 443` option. 

I don't really know where else to go from here. Also, stupid question, but what _exactly_ is the `sync` function? Does it actually download playable files, or does it download the proprietary video format?

Sorry for the stupid questions, and thanks for any help.

**Comment by danielfernau** [comment]
Hi @patatesboy 

This was caused by an issue within the source code, not by you. 🙂
A change to the 'ProtectClient' module in v2.0.1 without updating its usage in the 'sync' command resulted in the `unexpected keyword argument 'port'` error. I've just published an update that resolves this issue: [v2.0.2](https://github.com/unifi-toolbox/unifi-protect-video-downloader/releases/tag/v2.0.2)

Not a stupid question at all. The `sync` command is similar to the `download` command but it takes the date of the oldest available recording for each camera as the `--start` and downloads everything between that date and 'now' (instead of `--end`) to your destination folder. While doing so, the tool also keeps track of what's already been downloaded during a previous run. It's designed in a way that it can be used in an automation (e.g. a cron job or scheduled task) due to the fact that the command line stays the same since you only need to provide address, username, password and a destination, but no (constantly changing) start and end dates/times.

The files downloaded by the `download`, `events` and `sync` modules are always normal video files which can be opened using any common media player.

If you have any other questions feel free to post them here: [Discussions/Q&A](https://github.com/unifi-toolbox/unifi-protect-video-downloader/discussions/categories/q-a)
